web/src/overlays.ts

```diff
@@ -204,3 +204,93 @@ export function initialize(): void {
         e.stopPropagation();
     });
 }
+
+export function trap_focus_for_settings_overlay(): void {
+    $("#settings_overlay_container").on("keydown", (e) => {
+        if (e.key !== "Tab") {
+            return;
+        }
+
+        const two_column_mode = $("#settings_content").css("--single-column") === undefined;
+        const $settings_overlay_container = $("#settings_overlay_container");
+        let visible_focusable_elements;
+        if (two_column_mode) {
+            visible_focusable_elements =
+                overlay_util.get_visible_focusable_elements_in_overlay_container(
+                    $settings_overlay_container,
+                );
+        } else {
+            const $right_section = $settings_overlay_container.find(".content-wrapper");
+            const $left_section = $settings_overlay_container.find(".sidebar-wrapper");
+            if ($right_section.hasClass("show")) {
+                const $settings_panel = $right_section.find(".settings-section.show");
+                visible_focusable_elements =
+                    overlay_util.get_visible_focusable_elements_in_overlay_container(
+                        $settings_panel,
+                    );
+            } else {
+                visible_focusable_elements =
+                    overlay_util.get_visible_focusable_elements_in_overlay_container($left_section);
+            }
+        }
+
+        if (visible_focusable_elements.length === 0) {
+            return;
+        }
+
+        if (e.shiftKey) {
+            if (document.activeElement === visible_focusable_elements[0]) {
+                e.preventDefault();
+                visible_focusable_elements.at(-1)!.focus();
+            }
+        } else {
+            if (document.activeElement === visible_focusable_elements.at(-1)) {
+                e.preventDefault();
+                visible_focusable_elements[0]!.focus();
+            }
+        }
+    });
+
+    $("#channels_overlay_container, #groups_overlay_container").on("keydown", (e) => {
+        if (e.key !== "Tab") {
+            return;
+        }
+
+        const $overlay = $(e.currentTarget);
+        const two_column_mode =
+            $overlay.find(".two-pane-settings-container").css("--single-column") === undefined;
+        let visible_focusable_elements;
+        if (two_column_mode) {
+            visible_focusable_elements =
+                overlay_util.get_visible_focusable_elements_in_overlay_container($overlay);
+        } else {
+            const $right_section = $overlay.find(".right");
+            const $left_section = $overlay.find(".left");
```

Comment:
I guess we can deduplicate this if we later change how right/left work to align better between the two types of overlays.